edb/pgsql/compiler/expr.py

```diff
@@ -315,6 +315,24 @@ def compile_SliceIndirection(
     return result
 
 
+def _compile_call_args(
+    expr: irast.Call, *,
+    ctx: context.CompilerContextLevel
+) -> List[pgast.BaseExpr]:
+    args = [dispatch.compile(a.expr, ctx=ctx) for a in expr.args]
+    for ref, ir_arg, typemod in zip(args, expr.args, expr.params_typemods):
+        if (
+            not expr.impl_is_strict
+            and ir_arg.cardinality.can_be_zero()
+            and ref.nullable
+            and typemod == ql_ft.TypeModifier.SingletonType
+        ):
+            raise RuntimeError(
```

Comment:
Is this a temporary assertion?  If not, then this should be `errors.UnsupportedFeatureError`

Reply:
I made it a RuntimeError because that's what all the similar errors are, but they should probably all be UnsupportedFeatureError

Reply:
Can this surface in a user query? If so, a RuntimeErorr would become an ISE. Maybe we cleanup non-ISE RuntimeError in this file in a separate commit as part of this PR?